client/my-sites/pages/page-list.jsx

```diff
@@ -203,37 +203,40 @@ var Pages = React.createClass( {
 
 		// pages have loaded, sites have loaded, and we have a site instance or are viewing all-sites
 		if ( pages.length && this.props.sites.initialized ) {
-			if ( ! this.props.search ) {
-				// we're listing in reverse chrono. use the markers.
-				pages = this._insertTimeMarkers( pages );
-			}
-			rows = pages.map( function( page ) {
-				if ( ! ( 'site_ID' in page ) ) {
-					return page;
+			const site = this.props.sites.getSelectedSite();
+			const status = this.props.status || 'published';
+
+			// @TODO check if it is the first and last page (means there are fewer than 100)
+			if ( site && status === 'published' && ! this.props.search && this.props.lastPage ) {
```

Comment:
Could be good to add some semantics to this condition via better naming. Especially because of the chain of if/elses that follows.